libs/runner.js

```diff
@@ -0,0 +1,68 @@
+const replicator = require('./replicator'),
+      log = require('./log'),
+      env = require('../env')(),
+      xmlforms = require('./xmlforms'),
+      {delayLoop} = require('./delay');
+
+let firstRun = false;
+let errorCount = 0;
+let runTimes = 0;
+
+const replicateAll = async (couchUrl, pgconn) => {
+  let allResults = [],
+      runErrored = false;
+  try {
+    allResults = [
+      await replicator.replicator(couchUrl, pgconn).importAll(),
+      await replicator.replicator(`${couchUrl}-sentinel`, pgconn).importAll()
+    ];
+  } catch(err) {
+    log.error('Couch2PG import failed');
+    log.error(err);
+    if (err  && err.status === 401){
+      process.exit(1);
+    }
+    runErrored = true;
+  }
+  return [allResults, runErrored];
+};
+
+const run = async (couchUrl, pgconn, timesToRun=undefined) => {
+  log.info('Beginning couch2pg and xmlforms run at ' + new Date());
+  const [allResults, runErrored] = await replicateAll(couchUrl, pgconn);
+  if(allResults) {
+    const [medicResults, sentinelResults] = allResults;
+    // Run secondary tasks if we reasonably think there might be new data
+    // runErrored || errorCount <- something went wrong, but maybe there is still new data
+    // firstRun <- this is first run, we don't know what the DB state is in
+    // results.deleted.length || results.edited.length <- data has changed
```

Comment:
This comment is now out of date. If you believe it's self-explanatory delete the comment, or if you think it's a useful comment keep it up to date.